packages/main/src/ComboBox.ts

```diff
@@ -0,0 +1,1250 @@
+
+		if (isPhone() && this.value && !this._lastValue) {
+			this._lastValue = this.value;
+		}
+
+		this._toggleRespPopover();
+	}
+
+	_input(e: InputEvent) {
+		const { value } = e.target as HTMLInputElement;
+		const shouldAutocomplete = this.shouldAutocomplete(e);
+
+		if (e.target === this.inner) {
+			// stop the native event, as the semantic "input" would be fired.
+			e.stopImmediatePropagation();
+			this.focused = true;
+			this._isValueStateFocused = false;
+		}
+
+		this._filteredItems = this._filterItems(value);
+
+		this.value = value;
+		this.filterValue = value;
+
+		this._clearFocus();
+
+		// autocomplete
+		if (shouldAutocomplete && !isAndroid()) {
+			const item = this._getFirstMatchingItem(value);
+			item && this._applyAtomicValueAndSelection(item, value, true);
+
+			if (value !== "" && !this._selectionChanged && (item && !item.selected && !item.isGroupItem)) {
+				this.fireEvent<ComboBoxSelectionChangeEventDetail>("selection-change", {
+					item,
+				});
+
+				this._selectionChanged = false;
+			}
+		}
+
+		this.fireEvent("input");
+
+		if (isPhone()) {
+			return;
+		}
+
+		if (!this._filteredItems.length || value === "") {
+			this._closeRespPopover();
+		} else {
+			this._openRespPopover();
+		}
+	}
+	shouldAutocomplete(e: InputEvent): boolean {
+		const eventType: string = e.inputType;
+		const allowedEventTypes: Array<string> = [
+			"deleteWordBackward",
+			"deleteWordForward",
+			"deleteSoftLineBackward",
+			"deleteSoftLineForward",
+			"deleteEntireSoftLine",
+			"deleteHardLineBackward",
+			"deleteHardLineForward",
+			"deleteByDrag",
+			"deleteByCut",
+			"deleteContent",
+			"deleteContentBackward",
+			"deleteContentForward",
+			"historyUndo",
+		];
+
+		return !allowedEventTypes.includes(eventType);
+	}
+
+	_startsWithMatchingItems(str: string): Array<IComboBoxItem> {
+		return Filters.StartsWith(str, this._filteredItems, "text");
+	}
+
+	_clearFocus() {
+		this._filteredItems.map(item => {
+			item.focused = false;
+
+			return item;
+		});
+	}
+
+	handleNavKeyPress(e: KeyboardEvent) {
+		if (this.focused && (isHome(e) || isEnd(e)) && this.value) {
+			return;
+		}
+
+		const isOpen = this.open;
+		const currentItem = this._filteredItems.find(item => {
+			return isOpen ? item.focused : item.selected;
+		});
+
+		const indexOfItem = currentItem ? this._filteredItems.indexOf(currentItem) : -1;
+
+		e.preventDefault();
+
+		if (this.focused && isOpen && (isUp(e) || isPageUp(e) || isPageDown(e))) {
+			return;
+		}
+
+		if (this._filteredItems.length - 1 === indexOfItem && isDown(e)) {
+			return;
+		}
+
+		this._isKeyNavigation = true;
+
+		if (
+				 e.key === "ArrowDown"
+			|| e.key === "ArrowUp"
+			|| e.key === "PageUp"
+			|| e.key === "PageDown"
+			|| e.key === "Home"
+			|| e.key === "End"
+		) {
+			this[`_handle${e.key}`](e, indexOfItem);
+		}
+	}
+
+	_handleItemNavigation(e: KeyboardEvent, indexOfItem: number, isForward: boolean) {
+		const isOpen: boolean = this.open;
```

Comment:
in some function you set the var types explicitly in some you rely on the ts type inference

Reply:
I will remove redundant definitions